controllers/lmes/driver/driver.go

```diff
@@ -283,6 +283,29 @@ func (dc *driverComm) notifyShutdownWait() {
 	dc.connection <- 1
 }
 
+type PodLogAndPipeWriter struct {
+	pw *io.PipeWriter
+}
+
+func (f PodLogAndPipeWriter) Write(p []byte) (n int, err error) {
+	// format lm-eval output to stdout and the pipewriter
+	line := string(p)
+	res, _ := fmt.Print(line)
+	if _, err := f.pw.Write(p); err != nil {
+		return 0, err
+	}
+
+	// carriage returns do not correctly display, so replace with newlines
+	// carriage returns also break the pipewriter, so make sure we get newlines as well.
+	if strings.ContainsAny(line, "\r") && !strings.ContainsAny(line, "\n") {
```

Comment:
I am thinking would it be more straightforward to directly do a `regexp.ReplaceAll()` for all incoming []byte i.e:
```
re := regexp.MustCompile(`(?m)\r[^\n]`)
updatedBytes := re.ReplaceAll(p, []byte("\n"))
```

Reply:
@yhwang I've opened https://github.com/trustyai-explainability/trustyai-service-operator/issues/359 so this could be done on a separate PR after this one is tested, if that's ok.

Reply:
sounds good to me